caikit/core/task.py

```diff
@@ -241,6 +243,20 @@ def get_output_type(cls, output_streaming: bool) -> Type[DataBase]:
             raise ValueError("No streaming outputs are specified for this task")
         return cls.__annotations__[_STREAM_OUT_ANNOTATION]
 
+    @classmethod
+    def get_visibility(cls) -> bool:
+        """Get the visibility for this task.
+
+        NOTE: defaults to True even if visibility wasn't provided"""
+        return cls.__annotations__.get(_VISIBLE_ANNOTATION, True)
+
+    @classmethod
+    def get_extra_openapi_schema(cls) -> Dict[str, Any]:
```

Comment:
>  I wonder if it would make more sense to rephrase this as a more generic metadata property that could be interpreted as openapi in the http_server?

I think that makes sense, and we could define a `openapi_schema` key in the HTTP server to keep metadata generic